airline-core/pom.xml

```diff
@@ -27,6 +27,12 @@
       <version>${project.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>com.github.rvesse</groupId>
+      <artifactId>airline-backcompat-javaxinject</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
```

Comment:
I wonder if you should remove `jakarta.inject-api` below entirely from the classpath by default now?
The `@AirlineModule` thing is really nice and by default, I would get rid of all `@Inject`.

And people can reintroduce it for backward compatibility if they want.

Reply:
Yes in an ideal world I would do that but I'm aware there's a relatively large number of projects using Airline and until 3.x is ready I don't want to make unnecessary breaking changes.  Dropping `@Inject` would break all existing users on upgrades.

I've added lots of documentation indicating that support for both forms is a temporary measure and plan to fairly rapidly deprecate `@Inject` support over subsequent releases.